services/ui_backend_service/api/admin.py

```diff
@@ -0,0 +1,185 @@
+                    $ref: '#/definitions/ResponsesNotificationList'
+            "405":
+                description: invalid HTTP Method
+        """
+        processed_notifications = []
+        for notification in self.notifications:
+            try:
+                if "message" not in notification:
+                    continue
+
+                # Created at is required and "start" is used by default if not value provided
+                # Notification will be ignored if both "created" and "start" are missing
+                created = notification.get("created", notification.get("start", None))
+                if not created:
+                    continue
+
+                processed_notifications.append({
+                    "id": notification.get("id", hashlib.sha1(
+                        str(notification).encode('utf-8')).hexdigest()),
+                    "type": notification.get("type", "info"),
+                    "contentType": notification.get("contentType", "text"),
+                    "message": notification.get("message", ""),
+                    "url": notification.get("url", None),
+                    "urlText": notification.get("urlText", None),
+                    "created": created,
+                    "start": notification.get("start", None),
+                    "end": notification.get("end", None)
+                })
+            except:
+                pass
+
+        # Filter notifications based on query parameters
+        # Supports eq,ne.lt,le,gt,ge operators for all the fields
+        def filter_notifications(notification):
+            comp_operators = {
+                "eq": lambda a, b: a == b,
+                "ne": lambda a, b: a != b,
+                "lt": lambda a, b: a < b,
+                "le": lambda a, b: a <= b,
+                "gt": lambda a, b: a > b,
+                "ge": lambda a, b: a >= b,
+            }
+
+            try:
+                for q in request.query.keys():
+                    if ":" in q:
+                        field, op = q.split(":", 1)
+                    else:
+                        field, op = q, "eq"
+
+                    # Make sure compare operator is supported, otherwise ignore
+                    # Compare value is typecasted to match field type
+                    if op in comp_operators:
+                        field_val = notification.get(field, None)
+                        if not field_val:
+                            continue
+                        comp_val = type(field_val)(request.query.get(q, None))
+                        if not comp_val:
+                            continue
+                        if not comp_operators[op](field_val, comp_val):
+                            return False
+            except:
+                pass
+
+            return True
+
+        return web_response(status=200, body=list(
+            filter(filter_notifications, processed_notifications)))
+
+
+def _get_json_from_env(variable_name: str):
```

Comment:
move this function to utils

Reply:
#165 moves this to `ui_backend_service/api/utils.py`